src/py2spack/loading.py

```diff
@@ -0,0 +1,266 @@
+"""Utils for downloading packages from PyPI and opening the pyproject.tomls."""
+
+import io
+import tarfile
+from typing import Any, Dict, List, Optional
+
+import requests  # type: ignore
+import tomli
+from packaging import version as vn
+
+KNOWN_ARCHIVE_FORMATS = [
+    ".tar",
+    ".tar.gz",
+    ".tar.bz2",
+    ".gz",
+    ".xz",
+    ".bz2",
+]
+
+
+class APIError(Exception):
+    """Error during version lookup or querying of the PyPI JSON API."""
+
+    def __init__(
+        self,
+        msg: str,
+    ):
+        """Initialize error."""
+        super().__init__(msg)
+
+
+def _acceptable_version(version: str) -> Optional[vn.Version]:
+    """Check whether version string can be parsed and has correct format."""
+    try:
+        v = vn.parse(version)
+        # do not support post releases of prereleases etc.
+        if v.pre and (v.post or v.dev or v.local):
+            return None
+        return v
+    except vn.InvalidVersion:
+        return None
```

Comment:
In general the control flow of the code should be as linear as possible to make it easier to understand:  
```suggestion
    result = None
    try:
        v = vn.parse(version)
        # do not support post releases of prereleases etc.
        if not (v.pre and (v.post or v.dev or v.local)):
            result = v
    except vn.InvalidVersion:
        pass

    return result
```